gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactorJobRunner.java

```diff
@@ -397,4 +418,97 @@ public Status status() {
   public int compareTo(MRCompactorJobRunner o) {
     return Double.compare(o.priority, this.priority);
   }
+
+  private List<Path> getCumulativeLateFilePaths(Path lateDataDir) throws FileNotFoundException, IOException {
+    if (!this.fs.exists(lateDataDir)) {
+      return Lists.newArrayList();
+    }
+    List<Path> paths = Lists.newArrayList();
+    for (FileStatus fileStatus : FileListUtils.listFilesRecursively(fs, lateDataDir, new PathFilter() {
+      @Override
+      public boolean accept(Path path) {
+        for (String validExtention : getApplicableFileExtensions()) {
+          if (path.getName().endsWith(validExtention)) {
+            return true;
+          }
+        }
+        return false;
+      }
+    })) {
+      paths.add(fileStatus.getPath());
+    }
+    return paths;
+  }
+
+  /**
+   * Submit an event reporting new and cumulative late event counts.
+   */
+  private void submitLateEvent(List<Path> newLateFilePaths, Path lateDataDir) {
```

Comment:
Can your rename to something like submitLateEventsTrackingEvent or similar? The current name is a bit misleading.